src/calls/components/CallsView/Calendar.jsx

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import {
+    startOfMonth,
+    getDaysInMonth,
+    getDay,
+    getYear,
+    format,
+    addMonths,
+    subMonths,
+    addDays,
+    setMonth,
+    setYear
+} from 'date-fns';
+
+class Calendar extends Component {
+    state = {
+        date: new Date(),
+        months: [
+            'Jan',
+            'Feb',
+            'Mar',
+            'Apr',
+            'Maj',
+            'Jun',
+            'Jul',
+            'Avg',
+            'Sep',
+            'Okt',
+            'Nov',
+            'Dec'
+        ],
+        years: []
+    };
+
+    componentDidMount() {
+        this.props.handleChangeTableData(this.state.date);
+
+        if (this.state.years.length === 0) {
+            let years = [];
+            for (let y = 2000; y <= getYear(new Date()); y++) {
+                years.push(y);
+            }
+            this.setState({ years: years });
+        }
+    }
+
+    handleOnDateClick = event => {
+        let clickedDay = event.target.textContent;
+        let currentDate = this.state.date;
```

Comment:
This is a good place to use more idiomatic JavaScript. Normally, if we wanted to pull stuff out of state, we would write:

```javascript
const date = { this.state };
```

Now, if we want to call this variable `currentDate` in this scope, then that begs the question—why isn't the variable called `currentDate` in the state as well? Does it represent the current date? If yes, then it shouldn't really change. If it represents the selected date, then it could be called `selectedDate`.